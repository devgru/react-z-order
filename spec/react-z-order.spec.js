import React from 'react';
import ZOrder from '../lib';

import renderer from 'react-test-renderer';

describe('react-z-order', () => {
  it('should render at all', function() {
    const component = renderer.create(
      <ZOrder />
    );
    expect(component.toJSON()).toMatchSnapshot();
  });

  it('should sort', () => {
    const component = renderer.create(
      <ZOrder>
        {
          [10, 20, 40, 30].map(i => <circle x={i} z={i} key={i} />)
        }
      </ZOrder>
    );
    expect(component.toJSON()).toMatchSnapshot();
  });

  it('should be stable', () => {
    const component = renderer.create(
      <ZOrder>
        <circle x={10} z={10} />
        <circle x={30} z={10} />
        <circle x={20} z={10} />
      </ZOrder>
    );
    expect(component.toJSON()).toMatchSnapshot();
  });

  it('should throw when no Z specified', () => {
    expect(() => {
      renderer.create(
        <ZOrder>
          <circle x={40} />
          <circle x={10} z={10} />
        </ZOrder>
      );
    }).toThrowError(Error);
  });
});
