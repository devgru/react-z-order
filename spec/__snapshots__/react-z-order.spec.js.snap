exports[`react-z-order should be stable 1`] = `
<g>
  <circle
    x={10} />
  <circle
    x={30} />
  <circle
    x={20} />
</g>
`;

exports[`react-z-order should render at all 1`] = `<g />`;

exports[`react-z-order should sort 1`] = `
<g>
  <circle
    x={10} />
  <circle
    x={20} />
  <circle
    x={30} />
  <circle
    x={40} />
</g>
`;
